src/components/TransactionForm.tsx

```diff
@@ -313,18 +314,25 @@ const TransactionForm = ({ budget, setNotification }: Props) => {
                     destinationAccountId: account.id,
                   }
 
-                  // Suggest the first of the recentEnvelopes as the Envelope
-                  // to use for this transaction
-                  if (
-                    account.external &&
-                    !transaction.envelopeId &&
-                    account.recentEnvelopes?.length
-                  ) {
-                    setRecentEnvelopes(account.recentEnvelopes)
-                    valuesToUpdate.envelopeId = account.recentEnvelopes[0]
+                  let envelopePromises = []
+
+                  // Verify if we need to propose an envelope
+                  if (account.external && !transaction.envelopeId) {
+                    // Fetch the recent envelopes and suggest the first one
+                    // as the Envelope to use for this transaction
+                    envelopePromises.push(
+                      get(account.links.recentEnvelopes).then(data => {
+                        setRecentEnvelopes(data)
+                        if (data.length) {
+                          valuesToUpdate.envelopeId = data[0].id
+                        }
+                      })
+                    )
                   }
```

Comment:
We should probably also reset the recent envelopes if an internal account is chosen?
```suggestion
                  } else if (!account.external) {
                    setRecentEnvelopes([])
                  }
```